gear/gear/auth.py

```diff
@@ -12,110 +12,79 @@
 from hailtop.auth import async_get_userinfo
 from hailtop.config import get_deploy_config
 
+from .time_limited_max_size_cache import TimeLimitedMaxSizeCache
+
 log = logging.getLogger('gear.auth')
 
 deploy_config = get_deploy_config()
 
 BEARER = 'Bearer '
 
+TEN_SECONDS_IN_NANOSECONDS = int(1e10)
+
 
 def maybe_parse_bearer_header(value: str) -> Optional[str]:
     if value.startswith(BEARER):
         return value[len(BEARER) :]
     return None
 
 
-async def _userdata_from_session_id(session_id: str, client_session: httpx.ClientSession):
-    try:
-        return await async_get_userinfo(
-            deploy_config=deploy_config, session_id=session_id, client_session=client_session
+class AuthClient:
+    def __init__(self):
+        self._userdata_cache = TimeLimitedMaxSizeCache(
```

Comment:
I'm disappointed pylint didn't catch the type error. `TimeLimitedMaxSizeCache` has the right type parameters. Maybe we're not linting gear? Or maybe it's not smart enough to figure out the type parameters from `_load_userdata`?

Reply:
My assumption is that it's because it has no idea what type `session['session_id']` is, so it could have been a tuple for all it knows